db/migrate/20230815155824_validate_foreign_key_on_chances_of_successes.rb

```diff
@@ -0,0 +1,5 @@
+class ValidateForeignKeyOnChancesOfSuccesses < ActiveRecord::Migration[7.0]
+  def change
+    validate_foreign_key :chances_of_successes, :proceedings
```

Comment:
Could this be part of the first migration or does it need to be separate?

Reply:
@colinbruce This comes from a [strong_migrations recommendation](https://github.com/ankane/strong_migrations#good-10). 

Effectively because both tables are write locked while the new FK constraint is added, and validation make it more time consuming, they recommend adding the constraint without validating then validating separately. `

I suppose there is a minor risk of the validation migration failing - have not seen locally, or on the branch, I migrated with a prod dump on it - leaving some FKs being invalid but it otherwise seemed like a sensible approach to me.

Reply:
incidentally it took it 4.7 seconds to run the migrations on ~25k records.

Reply:
Cool :+1: